skyportal/tests/api/test_gcn.py

```diff
@@ -714,3 +720,231 @@ def test_gcn_summary_observations(
+            assert data["dateobs"] == "2019-08-14T21:10:39"
+            assert data["localization_name"] == "LALInference.v1.fits.gz"
+            assert np.isclose(np.sum(data["flat_2d"]), 1)
+            break
+        else:
+            time.sleep(2)
+    assert n_times_2 < 25
+
+    obj_id = str(uuid.uuid4())
+    status, data = api(
+        "POST",
+        "sources",
+        data={
+            "id": obj_id,
+            "ra": 24.6258,
+            "dec": -32.9024,
+            "redshift": 3,
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+
+    status, data = api("GET", f"sources/{obj_id}", token=view_only_token)
+    assert status == 200
+
+    status, data = api(
+        'POST',
+        'photometry',
+        data={
+            'obj_id': obj_id,
+            'mjd': 58709 + 1,
+            'instrument_id': ztf_camera.id,
+            'flux': 12.24,
+            'fluxerr': 0.031,
+            'zp': 25.0,
+            'magsys': 'ab',
+            'filter': 'ztfg',
+            "ra": 24.6258,
+            "dec": -32.9024,
+            "ra_unc": 0.01,
+            "dec_unc": 0.01,
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    params = {
+        "sourcesIdList": obj_id,
+    }
+    status, data = api(
+        'GET',
+        'sources_in_gcn/2019-08-14T21:10:39',
+        params=params,
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert len(data["data"]) == 0
+
+    # confirm source
+    params = {
+        "sourceId": obj_id,
+        "localizationName": "LALInference.v1.fits.gz",
+        "localizationCumprob": 0.95,
+        "confirmedOrRejected": True,
+        "startDate": "2019-08-13 08:18:05",
+        "endDate": "2019-08-19 08:18:05",
+    }
+
+    # verify that you can't confirm a source without the Manage GCNs permission
+    status, data = api(
+        'POST',
+        'sources_in_gcn/2019-08-14T21:10:39',
+        data=params,
+        token=upload_data_token,
+    )
+    assert status == 401
+
+    status, data = api(
+        'POST',
+        'sources_in_gcn/2019-08-14T21:10:39',
+        data=params,
+        token=super_admin_token,
+    )
+    assert status == 200
+
+    params = {
+        "sourcesIdList": obj_id,
+    }
+    status, data = api(
+        'GET',
+        'sources_in_gcn/2019-08-14T21:10:39',
+        params=params,
+        token=upload_data_token,
+    )
+    assert status == 200
+    data = data["data"]
+    assert len(data) == 1
+    assert data[0]["obj_id"] == obj_id
+    assert data[0]["dateobs"] == "2019-08-14T21:10:39"
+    assert data[0]["confirmed_or_rejected"] is True
+
+    # find gncs associated to source
+    status, data = api(
+        'GET',
+        f"source_in_gcns/{obj_id}",
+        token=upload_data_token,
+    )
+    assert status == 200
+    data = data["data"]
+    assert '2019-08-14T21:10:39' in data['gcns']
```

Comment:
This PR actually adds that to the source page! On the source page, if the source has been confirmed as the one being at the origin of an event, it will display in the InfoList the GCN (or GCNs) associated to the source, as clickable links. It's something users asked me for. That once a source is confirmed as being the one at the origin of a GCN, it is also mentioned on the source page, so they know that if a source already has been confirmed in a GCN when looking at it.
If you think it doesn't make sense, I can simply remove it it's not a problem for me. Maybe @mcoughlin can tell us what he thinks? He knows about that feature.

Now why GCNs and not just one GCN. Well technically in the table I added, a source can be confirmed in different events. So I had to list all of them rather than just one, which might miss a gcn is was confirmed in. If you think that a source can be confirmed for only ONE GCN, then I can add a unique constraint on a `confirmed` source, but not unique on a `rejected` source. Not sure how to do that, but it would be that the source + confirmed being true has to be unique in the DB ? Or at least doing that in the handler.

Reply:
I think you can have multiple GCNs in case of a GRB and GW from the same neutron star merger. This gets at the question of aliases that @Theodlz and @leoyang33 are tackling.

Reply:
Yeah it is true that in principle it can be associated with more than one GCN. We just want to design around thinking it would probably not be a lot of associations. For example, you may have an easier time making something like classifications (where you have space for a few, not a whole table).

Reply:
What do you mean by "something like the classifications"? I'm looking at both frontend and backend of the classifications, and it seems that it is querying and displaying all the classifications associated to a source, not just a few. The only thing that is done for the classifications is that it prevents duplicates in the frontend(for every identical classification, it shows the latest one), but it does show all of them.